src/descriptor/satisfied_constraints.rs

```diff
@@ -200,9 +201,9 @@ pub enum SatisfiedConstraint<'desc, 'stack> {
 ///the top of the stack, we need to decide whether to execute right child or not.
 ///This is also useful for wrappers and thresholds which push a value on the stack
 ///depending on evaluation of the children.
-struct NodeEvaluationState<'desc> {
+struct NodeEvaluationState<'desc, Ctx: 'desc + ScriptContext> {
     ///The node which is being evaluated
-    node: &'desc Miniscript<bitcoin::PublicKey>,
+    node: &'desc Miniscript<bitcoin::PublicKey, Ctx>,
```

Comment:
I think we should put `Any` here rather than genericizing over `Ctx` and then using `Any` down below.